pkg/chain/ethereum/ethereum_integration_test.go

```diff
@@ -0,0 +1,85 @@
+//go:build integration
+// +build integration
+
+package ethereum
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/keep-network/keep-common/pkg/chain/ethereum/ethutil"
+	"github.com/keep-network/keep-core/pkg/internal/testutils"
+	"reflect"
+	"testing"
+	"time"
+)
+
+// TODO: Include integration test in the CI.
+// To run the tests execute `go test -v -tags=integration ./...`
```

Comment:
I do not feel comfortable merging it without having the code for `GetBlockNumberByTimestamp` covered in CI and it may take a moment to implement integration tests action right. :( I think we could quite easily rework it into a unit test though.

One option is to implement `_getBlockNumberByTimestamp` that will accept two functions as parameters: one to get the current block and another to get the block by number. Then, `GetBlockNumberByTimestamp` would call it. This approach would let us implement even more tests - we can generate any array of blocks we want.

Another option is to instantiate `baseChain` manually passing a mock implementation of `client` and `blockCounter` but I feel it is more work and boilerplate code.

Reply:
Yeah, I was wondering about integration vs unit tests for that. I chose the former because:
- it makes more sense to test this function with a real ETH client, especially performance-wise
- `blockByNumber` is not straightforward to mock as their return type `*types.Block` holds the relevant data as private fields. It is possible to overcome this by adding additional interfaces but that seemed non-elegant to me and hence I decided to stick with integration tests